File: golang/task1/task1.go
package main

import (
	"fmt"
	"github.com/tealeg/xlsx"
	"net/http"
	_ "path/filepath"
	handle_function "pham-hong-dang-shub-test/golang/task1/handle-function"
)

var uploadedFile *xlsx.File

func main() {
	http.HandleFunc("/", handle_function.HandleIndex)
	http.HandleFunc("/query", handle_function.HandleQuery)
	http.HandleFunc("/api/upload", func(writer http.ResponseWriter, request *http.Request) {
		uploadedFile = handle_function.HandleUpload(writer, request, uploadedFile)
	})
	http.HandleFunc("/api/query", func(writer http.ResponseWriter, request *http.Request) {
		handle_function.HandleApiQuery(writer, request, uploadedFile)
	})

	fmt.Println("Server is running on port 3000")
	http.ListenAndServe(":3000", nil)
}

File: node-js/task1/task1.js
const express = require('express');
const app = express();
const port = 3000;

const multer = require('multer');
const upload = multer({dest: 'uploads/'});

const handleIndex = require("./handle-function/handle-index");
const handleQuery = require("./handle-function/handle-query");
const handleUpload = require("./handle-function/handle-upload");
const handleApiQuery = require("./handle-function/handle-api-query");

let uploadedFile = null;

app.get('/', handleIndex);

app.get('/query', handleQuery);

app.post('/api/upload', upload.single('file'), (req, res) => {
    uploadedFile = handleUpload(req, res);
});

app.get('/api/query', (req, res) => {
    handleApiQuery(req, res, uploadedFile);
});

app.listen(port, () => {
    console.log(`Server is running on port ${port}`);
});